tests/blas_interface_test_gtest.hpp

```diff
@@ -0,0 +1,121 @@
+#include <gtest/gtest.h>
+
+using test_t = ::testing::tuple<QudaBLASType, QudaBLASDataType>;
+
+class BLASTest : public ::testing::TestWithParam<test_t>
+{
+protected:
+  test_t param;
+
+public:
+  BLASTest() : param(GetParam()) { }
+  // virtual ~BLASTest() { }
+};
+
+bool skip_test(test_t param)
+{
+  if (::testing::get<0>(param) == QUDA_BLAS_LU_INV
+      && (::testing::get<1>(param) == QUDA_BLAS_DATATYPE_D || ::testing::get<1>(param) == QUDA_BLAS_DATATYPE_S))
+    return true;
+  else
+    return false;
+}
+
+// For googletest, names must be non-empty, unique, and may only contain ASCII
+// alphanumeric characters or underscore.
+const char *data_type_str[] = {
+  "realSingle",
+  "realDouble",
+  "complexSingle",
+  "complexDouble",
+};
+
+const char *test_type_str[] = {
+  "GEMM",
+  "LUInvert",
+};
+
+// Helper function to construct the test name
+std::string getBLASDataName(testing::TestParamInfo<test_t> param)
+{
+  auto data_type = ::testing::get<1>(param.param);
+  std::string str(data_type_str[data_type]);
+  return str;
+}
+
+// Helper function to construct the test name
+std::string getBLASTestName(testing::TestParamInfo<test_t> param)
+{
+  auto test_type = ::testing::get<0>(param.param);
+  std::string str(test_type_str[test_type]);
+  return str;
+}
+
+// The following tests gets each BLAS type and precision using google testing framework
+using ::testing::Bool;
+using ::testing::Combine;
+using ::testing::Range;
+using ::testing::TestWithParam;
+using ::testing::Values;
+
+double gemm_test(test_t test_param);
+double lu_inv_test(test_t test_param);
+
+// Sets up the Google test
+TEST_P(BLASTest, verify)
+{
+  if (skip_test(GetParam())) GTEST_SKIP();
+
+  auto param = GetParam();
+  auto test_type = ::testing::get<0>(param);
+  auto data_type = ::testing::get<1>(param);
+  switch (test_type) {
+  case QUDA_BLAS_GEMM: {
+    auto deviation_gemm = gemm_test(param);
+    decltype(deviation_gemm) tol_gemm;
+    switch (data_type) {
+    case 0:
```

Comment:
Can `QUDA_BLAS_DATATYPE_S`, etc, be used here to avoid the "magic number" 0, 1, 2, 3?

Reply:
Swapped both instances, good catch. a43b61988